### PR TITLE
Upgrade to Scala 2.11.12

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

**Comment by dwijnand** [comment]
Re review @chbatey?

File: build.sbt [hunk 1]
```diff
@@ -119,7 +119,8 @@ lazy val `akka-discovery-aws-api-async` = project
   .settings(
     name := "akka-discovery-aws-api-async",
     organization := "com.lightbend.akka.discovery",
-    Dependencies.DiscoveryAwsApiAsync
+    Dependencies.DiscoveryAwsApiAsync,
+    scalacOptions += "-target:jvm-1.8"
```
[the remark this answers is not in the document]
Comment (dwijnand) [reply build.sbt]: ¯\_(ツ)_/¯ should we? to what?

[the remark this answers is not in the document]
Comment (raboof) [reply build.sbt]: If we don't, building on JDK9 could produce artifacts that don't work on JDK8... so I think we should?

We used `CrossJava` to find it in the Akka build (https://github.com/akka/akka/blob/ca08ca36e24df2d49850926aa78da3baad19df2d/project/AkkaBuild.scala#L102), we might want to apply the same trick here?

**Comment by dwijnand** [comment]
This doesn't seem worth it.

**Pull request closed without merging**